modules/configuration/src/main/java/org/apache/ignite/internal/configuration/util/ConfigurationNotificationsUtil.java

```diff
@@ -250,23 +366,25 @@ public static void notifyListeners(
         List<CompletableFuture<?>> futures,
         BiFunction<L, ConfigurationNotificationEvent<V>, CompletableFuture<?>> updater
     ) {
-        ConfigurationNotificationEvent<V> evt = new ConfigurationNotificationEventImpl<>(
-            oldVal,
-            newVal,
-            storageRevision
-        );
+        if (!listeners.isEmpty()) {
```

Comment:
Why not just add this?:
if (listeners.isEmpty())
    return;
This way code will much easier to read